executor/ddl.go

```diff
@@ -798,13 +798,15 @@ func (e *DDLExec) executeLockTables(s *ast.LockTablesStmt) error {
 	}
 
 	if !config.TableLockEnabled() {
+		e.ctx.GetSessionVars().StmtCtx.AppendWarning(errors.Errorf("Lock tables works only when enable-table-lock is set in config file"))
 		return nil
 	}
 	return domain.GetDomain(e.ctx).DDL().LockTables(e.ctx, s)
 }
 
 func (e *DDLExec) executeUnlockTables(_ *ast.UnlockTablesStmt) error {
 	if !config.TableLockEnabled() {
+		e.ctx.GetSessionVars().StmtCtx.AppendWarning(errors.Errorf("Unlock tables works only when enable-table-lock is set in config file"))
 		return nil
 	}
```

Comment:
A couple of suggestions:
1. We make the mode dependent on https://docs.pingcap.com/tidb/dev/system-variables#tidb_enable_noop_functions-new-in-v40 -- but it only ever returns warnings at most (since it's important DDL can restore). This provides a mechanism for users to kill the warning if that is problematic for their script.
2. We add a new error number / error code. This helps us track the errors in information schema so we know which unsupported syntax the user is encountering.

Let me know what you think about either. These are just suggestions.

Reply:
1. Sounds good for me.
2. Maybe we can use `mysql.ErrNotSupportedYet` with different error message between `ErrFunctionsNoopImpl`.

Reply:
Sounds good to me.

Reply:
Done，PTAL @morgo